keanu-project/src/main/java/io/improbable/keanu/templating/SequenceItem.java

```diff
@@ -66,12 +76,23 @@ public SequenceItem() {
         return copyOf(this.contents);
     }
 
+    public int getPosition() {
+        return this.itemPosition;
+    }
+
     private String getUniqueName() {
-        return NAME_PREFIX + this.hashCode();
+        return NAME_PREFIX + this.itemPosition;
     }
 
     private VertexLabel scoped(VertexLabel label) {
-        return label.withExtraNamespace(getUniqueName());
+        VertexLabel scopedLabel = label
+            .withExtraNamespace(String.valueOf(this.uniqueIdentifier))
+            .withExtraNamespace(getUniqueName());
+
+        if (this.identifyingNamespace != null) {
+            scopedLabel = scopedLabel.withExtraNamespace(this.identifyingNamespace);
```

Comment:
could you please add a test that checks you get an exception when you call `withExtraNamespace(null)`? It will pass because you'll get an exception from `ImmutableList.builder().add(null)`, I just want to make sure the test is there.

Reply:
So this is a test on a `VertexLabel`?